pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -146,11 +140,23 @@ func parseCRILog(log []byte, msg *logMessage) error {
 	if idx < 0 {
 		return fmt.Errorf("stream type is not found")
 	}
-	msg.stream = streamType(log[:idx])
-	if msg.stream != stdoutType && msg.stream != stderrType {
+	msg.stream = runtimeapi.StreamType(log[:idx])
+	if msg.stream != runtimeapi.Stdout && msg.stream != runtimeapi.Stderr {
 		return fmt.Errorf("unexpected stream type %q", msg.stream)
 	}
 
+	// Parse log tag
+	log = log[idx+1:]
+	idx = bytes.Index(log, delimiter)
+	if idx < 0 {
+		return fmt.Errorf("log tag is not found")
```

Comment:
Will this break old log format? e.g. a single line with no tag set.

Reply:
Yes, and each next log metadata change can do it too, that's why I push towards JSON format instead

Reply:
If we don't want to break current format, we could use the following format instead:
```
TIMESTAMP TAG CONTENT
2016-10-06T00:17:09.669794202Z stdout:P log content 1
```
It just depends on whether we think we should keep backward compatibility now.

And in the future, we could always extend the tag field with new sub field, if we really have to.

Reply:
And we could switch to json format in the future if we really need many metadata fields.